crates/swc_ecma_transforms_compat/src/es2017/async_to_generator.rs

```diff
@@ -192,27 +187,32 @@ impl Fold for Actual {
                 args,
                 type_args,
             }) if callee.is_fn_expr() => {
-                return self.handle_iife(span, (*callee).fn_expr().unwrap(), args, type_args);
+                *expr = self.handle_iife(
+                    span.take(),
+                    &mut (*callee).take().fn_expr().unwrap(),
+                    args,
+                    type_args.take(),
+                );
             }
 
             Expr::Call(CallExpr {
                 span,
                 callee: ExprOrSuper::Expr(callee),
                 args,
                 type_args,
-            }) if match &*callee {
+            }) if match &**callee {
                 Expr::Paren(ParenExpr { expr, .. }) => match &**expr {
                     Expr::Fn(..) => true,
                     _ => false,
                 },
                 _ => false,
             } =>
             {
-                return self.handle_iife(
-                    span,
-                    (*callee).paren().unwrap().expr.fn_expr().unwrap(),
+                *expr = self.handle_iife(
```

Comment:
Can you add `return` after this?

Reply:
Short curcuiting with return here actually makes some existing test fails, if fn is wrapped iife and it contains async.